gammapy/data/data_store.py

```diff
@@ -208,28 +219,30 @@ def obs(self, obs_id):
         if obs_id not in self.hdu_table["OBS_ID"]:
             raise ValueError(f"OBS_ID = {obs_id} not in HDU index table.")
 
+        available_IRF = ["aeff", "edisp", "psf", "bkg"]
+        if required_IRF is "all":
+            required_IRF = available_IRF
+        if required_IRF is None:
+            required_IRF = []
+
+        if not set(required_IRF).issubset(available_IRF):
+            difference = set(required_IRF).difference(available_IRF)
+            raise ValueError(f"{difference} is not a valid method.")
+
         row = self.obs_table.select_obs_id(obs_id=obs_id)[0]
-        obs_info = table_row_to_dict(row)
-
-        aeff_hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type="aeff")
-        edisp_hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type="edisp")
-        bkg_hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type="bkg")
-        psf_hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type="psf")
-        events_hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type="events")
-        gti_hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type="gti")
-
-        return Observation(
-            obs_id=int(obs_id),
-            obs_info=obs_info,
-            bkg=bkg_hdu,
-            aeff=aeff_hdu,
-            edisp=edisp_hdu,
-            events=events_hdu,
-            gti=gti_hdu,
-            psf=psf_hdu,
-        )
+        kwargs = {"obs_id": int(obs_id)}
+        kwargs["obs_info"] = table_row_to_dict(row)
+
+        for irf in required_IRF + ["events", "gti"]:
+            hdu = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type=irf)
+            if hdu is None:
+                log.warning(f"Skipping run with missing IRF obs_id: {obs_id!r}")
+                return None
+            kwargs[irf] = hdu
 
-    def get_observations(self, obs_id=None, skip_missing=False):
+        return Observation(**kwargs)
+
+    def get_observations(self, obs_id=None, skip_missing=False, required_IRF="all"):
```

Comment:
Rename to `required_irfs` (all lower-case and plural...). Or for consistency one could also choose `skip_missing_irfs=[]`...

In the loop you check the wether the corresponding IRF is `None`.